### PR TITLE
suggest `*.desktop` for AppImage

### PR DESCRIPTION
**Pull request by Rudxain**
Since the official AppImage doesn't include the icon/logo/isotype, and some desktop environments (such as GNOME) don't allow executing arbitrary files by double-clicking (for security reasons), I believe it's beneficial for the user to have the `.desktop` shortcut file

### CODE REVIEW

**Comment by Rudxain** [comment]
BTW, it would be nice if [shell completions](/helix-editor/helix/tree/master/contrib/completion) were documented, because AppImage and Nightly users may need it

**Comment by krish-r** [comment]
Minor suggestions: (Please feel free to ignore)

- It'll be better to avoid the temp variable (`$d`) to avoid any potential overwrites.
(Also, fish for ex. does not support `var=val` or `unset`)

- Instead of suggesting this as a pre-requisite for desktop shortcut, can this be something like - 
"To run / start Helix from anywhere, make sure to add it to your $PATH".

- Also, I think it'll be better to keep the same link text for consistency (might also be easy during the search) - 
"You can also optionally `Configure the desktop shortcut`

File: book/src/package-managers.md [hunk 1]
```diff
@@ -101,7 +101,15 @@ Download the official Helix AppImage from the [latest releases](https://github.c
 chmod +x helix-*.AppImage # change permission for executable mode
 ./helix-*.AppImage # run helix
 ```
- 
+
+You can optionally [add the `.desktop` file](./building-from-source.md#configure-the-desktop-shortcut), but there's some pre-requisites:
```
Comment (the-mikedavis) [inline book/src/package-managers.md:105]:
```suggestion
You can optionally [add the `.desktop` file](./building-from-source.md#configure-the-desktop-shortcut). Helix must be installed in `PATH` with the name `hx`. For example:
```

File: book/src/package-managers.md [hunk 2]
```diff
@@ -101,7 +101,15 @@ Download the official Helix AppImage from the [latest releases](https://github.c
...
+```sh
+mkdir -p "$HOME/.local/bin" && \
+mv helix-*.AppImage "$HOME/.local/bin/hx"
+```
+
+`hx` can be anywhere in your `$PATH`, but (by [convention](https://askubuntu.com/a/1408444)) `~/.local/bin` is preferred.
```
Comment (the-mikedavis) [inline book/src/package-managers.md:111]:
```suggestion
and make sure `~/.local/bin` is in your `PATH`.
```

justifying the use of `~/.local/bin` seems unnecessary/offtopic to me, especially if we frame the directions above as an example

File: book/src/package-managers.md [hunk 3]
```diff
@@ -101,7 +101,15 @@ Download the official Helix AppImage from the [latest releases](https://github.c
 chmod +x helix-*.AppImage # change permission for executable mode
 ./helix-*.AppImage # run helix
 ```
- 
+
+You can optionally [add the `.desktop` file](./building-from-source.md#configure-the-desktop-shortcut), but there's some pre-requisites:
+```sh
+mkdir -p "$HOME/.local/bin" && \
+mv helix-*.AppImage "$HOME/.local/bin/hx"
```
Comment (the-mikedavis) [inline book/src/package-managers.md:108]:
```suggestion
mkdir -p "$HOME/.local/bin"
mv helix-*.AppImage "$HOME/.local/bin/hx"
```

I think it's clearer if these are separate commands

Comment (Rudxain) [reply book/src/package-managers.md:108]: I guess? It looks cleaner without `&&`. I added `&&` for safety and performance, but I guess that's irrelevant here, especially considering `mv` should fail anyways if the dir is non-existent

Comment (the-mikedavis) [reply book/src/package-managers.md]: yeah that was my thinking as well, if `mkdir` fails then `mv` certainly will too

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**